src/operator/tensor/dot-inl.h

```diff
@@ -798,11 +776,16 @@ inline void DotCsrRspDnsImpl(const OpContext& ctx,
           if (trans_lhs) {
             LOG(FATAL) << "DotCsrRspDnsImpl has not implemented dot(csr.T, rsp) = dns yet";
           } else {
+            const RType* row_idx_ptr = row_idx_r.dptr<RType>();
+            std::unordered_map<RType, dim_t> row_idx_map;
+            for (dim_t ind = 0; ind < nnr; ind++) {
+              row_idx_map.emplace(row_idx_ptr[ind], ind);
+            }
             mxnet_op::Kernel<DotCsrRspDnsByRowBlocks, cpu>::Launch(s, num_threads,
```

Comment:
No need to parallelize by `RowBlocks`. 1 thread per row is fine and potentially leads to better workload balance